cli/azd/pkg/prompt/prompter.go

```diff
@@ -77,7 +77,7 @@ func (p *DefaultPrompter) EnsureEnv(ctx context.Context) error {
 			return err
 		}
 
-		p.env.SetLocation(location)
+		p.env.DotenvSet(environment.LocationEnvVarName, location)
```

Comment:
We should be consistent on how we set these well known variables.  `SetLocation(...)` already calls `DotenvSet(...)` with the same constant.  If we don't like the well known overrides like `SetLocation(...)` we should just deprecate them and update all the usages.